python/private/pypi/pip_compile.bzl

```diff
@@ -147,6 +157,12 @@ def pip_compile(
     # cheap way to detect the bazel version
     _bazel_version_4_or_greater = "propeller_optimize" in dir(native)
 
+    # setuptools (the default python build tool) attempts to find user configuration in the user's home direcotory. This seems to work fine on linux and macOS, but fails on Windows. We provide a fake USERPROFILE env variable to allow setuptools to proceed without finding user-provided configuration.
+    kwargs["env"] = kwargs.pop("env", {}) | select({
```

Comment:
huh? I thought `env` is a non-configurable attribute (at least it was in some other rules), if the CI is happy, I'll trust it.